history.go

```diff
@@ -54,11 +68,15 @@ func (h *History) UpdateHistory(policies Policies, m *MerkleTree, nextEp Epoch)
 		}
 		h.loadedEpochs = append(h.loadedEpochs[:0], h.loadedEpochs[n:]...)
 	}
-	h.snapshots[nextEp] = nextStr
-	h.loadedEpochs = append(h.loadedEpochs, nextEp)
+
+	h.updateInternal(policies, nextEp)
 	return nil
 }
 
+func (h *History) Set(key string, value []byte) error {
```

Comment:
It definitely it makes sense to have a wrapper around ``tree.Set()`` to abstract away the tree, but now that I think about it, isn't the semantics we want more like ``directory.Set()``? Similarly, below, wouldn't we want more something like ``directory.Lookup()``? If what we want is to abstract away the underlying PAD (including the history), shouldn't we only expose an interface to a directory at the server level?